firmware/applications/SignBuddy/imu.c

```diff
@@ -48,6 +49,20 @@ static void hw_init(void)
   i2c_init(&s.i2c_instance, IMU_I2C, &i2c_config);
 }
 
+static inline void sample_data(void)
+{
+  ulTaskNotifyTake(pdFALSE, portMAX_DELAY);
+  uint8_t ret = 0;
```

Comment:
You are calling task notify twice, once here and once on line 98

Reply:
yeah that'll mess the timing up good

thanks @stephen99scott